src/client/Microsoft.Identity.Client/AppConfig/CiamAuthorityHelper.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// Translates CIAM authorities into a usable form. This is needed only until ESTS is updated to support the north star format
+    /// North star format: https://idgciamdemo.ciamlogin.com
+    /// Transformed format: https://idgciamdemo.ciamlogin.com/idgciamdemo.onmicrosoft.com
+    /// </summary>
+    internal class CiamAuthorityHelper
+    {
+        public CiamAuthorityHelper(Uri authority)
+        {
+            TransformAuthority(authority);
+        }
+
+        public CiamAuthorityHelper(string instance, string tenant)
+        {
+            instance = instance.Trim().TrimEnd('/');
+            if (string.IsNullOrEmpty(tenant))
+            {
+                TransformAuthority(new Uri(instance + '/' + tenant));
+            }
+            else
+            {
+                TransformedInstance = instance;
+                TransformedTenant = tenant;
+                TransformedAuthority = new Uri(instance + '/' + tenant);
+            }
+        }
+
+        public Uri TransformedAuthority { get; private set; }
+
+        public string TransformedTenant { get; private set; }
+
+        public string TransformedInstance { get; private set; }
```

Comment:
I think these two can be made private, since they aren't used anywhere else; and they will be tested transitively though tests on TransformedAuthority.

Reply:
Yea, I was planning on using these at some point but I ended up not needing them.